argilla/src/argilla/records/_io/_datasets.py

```diff
@@ -26,6 +27,125 @@
     from argilla.datasets import Dataset
 
 
+def _cast_images_as_urls(hf_dataset: "HFDataset", columns: List[str]) -> "HFDataset":
+    """Cast the image features in the Hugging Face dataset as URLs.
+
+    Parameters:
+        hf_dataset (HFDataset): The Hugging Face dataset to cast.
+        repo_id (str): The ID of the Hugging Face Hub repo.
+
+    Returns:
+        HFDataset: The Hugging Face dataset with image features cast as URLs.
+    """
+
+    for column in columns:
+        # make an updated features object with the new column type
+        features = hf_dataset.features.copy()
+        features[column] = Value("string")
+        # cast the column in batches
+        hf_dataset = hf_dataset.map(
+            function=lambda batch: {column: [pil_to_data_uri(sample) for sample in batch]},
+            with_indices=False,
+            batched=True,
+            input_columns=[column],
+            remove_columns=[column],
+            features=features,
+        )
+
+    return hf_dataset
+
+
+def _cast_classlabels_as_strings(hf_dataset: "HFDataset", columns: List[str]) -> "HFDataset":
+    """Cast the class label features in the Hugging Face dataset as strings.
+
+    Parameters:
+        hf_dataset (HFDataset): The Hugging Face dataset to cast.
+        columns (List[str]): The names of the columns containing the class label features.
+
+    Returns:
+        HFDataset: The Hugging Face dataset with class label features cast as strings.
+    """
+    for column in columns:
+        features = hf_dataset.features.copy()
+        features[column] = Value("string")
+        hf_dataset = hf_dataset.map(
+            lambda x: {column: hf_dataset.features[column].int2str(x[column])}, features=features
+        )
+    return hf_dataset
+
+
+FEATURE_CASTERS = {
+    Image: _cast_images_as_urls,
+    ClassLabel: _cast_classlabels_as_strings,
+}
+
+
+def _uncast_uris_as_images(hf_dataset: "HFDataset", columns: List[str]) -> "HFDataset":
+    """Cast the image features in the Hugging Face dataset as PIL images.
+
+    Parameters:
+        hf_dataset (HFDataset): The Hugging Face dataset to cast.
+        columns (List[str]): The names of the columns containing the image features.
+
+    Returns:
+        HFDataset: The Hugging Face dataset with image features cast as PIL images.
+    """
+
+    for column in columns:
+        features = hf_dataset.features.copy()
+        features[column] = Image()
+        casted_hf_dataset = hf_dataset.map(
+            function=lambda batch: {column: [uncast_image(sample) for sample in batch]},
+            with_indices=False,
+            batched=True,
+            input_columns=[column],
+            remove_columns=[column],
+            features=features,
+        )
+        try:
+            casted_hf_dataset[0]
```

Comment:
is this correct? when a `None` return, 
```suggestion
        
        try:
            casted_hf_dataset = hf_dataset.map(
                function=lambda batch: {column: [uncast_image(sample) for sample in batch]},
                with_indices=False,
                batched=True,
                input_columns=[column],
                remove_columns=[column],
                features=features,
            )
            casted_hf_dataset[0]
```

Reply:
`casted_hf_dataset[0]` will fail if the image value is not compatible with dataset serialization. Why should the map execution be inside the exception?